packages/turbo/android/src/main/java/com/reactnativeturbowebview/RNSession.kt

```diff
@@ -154,6 +159,14 @@ class RNSession(
     topmostView?.visitRendered()
   }
 
+  override fun formSubmissionStarted(location: String) {
+    topmostView?.didStartFormSubmission(location)
+  }
+
+  override fun formSubmissionFinished(location: String) {
+    topmostView?.didFinishFormSubmission(location)
```

Comment:
I'm wondering what happens if the location of the `topmostView` has changed in the meantime, ie. due to `url` being updated or a different view has become the topmost one? Is this something we should care about and/or filter?

Reply:
When you change screen during form upload (submissionStarted callback is called and submissionFinished is not) then this new screen (== new topmostView) calls `submissionFinished` callback (it is not called on a screen where form has been submitted). So it appears that sessionDelegate is changed before `formSubmissionFinished` is called. So it might make sense to store what topmostView called `submissionStarted` just to avoid calling `formSubmissionFinished` on a screen which has not submitted a form.

Reply:
The corresponding code in `Turbo Navigator iOS` looks like [this](https://github.com/hotwired/turbo-ios/blob/turbo-navigator/Source/Turbo%20Navigator/TurboNavigator.swift#L108-L121):

```swift
    public func sessionDidStartFormSubmission(_ session: Session) {
        if let url = session.topmostVisitable?.visitableURL {
            delegate.formSubmissionDidStart(to: url)
        }
    }

    public func sessionDidFinishFormSubmission(_ session: Session) {
        if session == modalSession {
            self.session.clearSnapshotCache()
        }
        if let url = session.topmostVisitable?.visitableURL {
            delegate.formSubmissionDidFinish(at: url)
        }
    }
```

These seem fairly app-specific (ie. the `session.clearSnapshotCache()` stuff which belongs in the app-code.

I think triggering the `onFormSubmissionFinished` in the `VisitableView` without any filtering is what we want as long as we pass the URL, the corresponding handler can always ignore the call, if the URLs don't match.

We can consider adding some documentation regarding this, ie. that the form submission handlers are triggered for the _session_ and a `url` argument is available for granular control.